plugins/notifications-backend/src/service/NotificationCleaner.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  LoggerService,
+  readSchedulerServiceTaskScheduleDefinitionFromConfig,
+  SchedulerService,
+  SchedulerServiceTaskScheduleDefinition,
+} from '@backstage/backend-plugin-api';
+import { Config, readDurationFromConfig } from '@backstage/config';
+import { NotificationsStore } from '../database';
+import { durationToMilliseconds, HumanDuration } from '@backstage/types';
+
+export class NotificationCleaner {
+  static readonly initNotificationCleaner = async (
+    config: Config,
+    scheduler: SchedulerService,
+    logger: LoggerService,
+    database: NotificationsStore,
+  ): Promise<void> => {
+    const schedule: SchedulerServiceTaskScheduleDefinition = config.has(
+      'notifications.expiration.schedule',
+    )
+      ? readSchedulerServiceTaskScheduleDefinitionFromConfig(
+          config.getConfig('notifications.expiration.schedule'),
+        )
+      : {
+          frequency: { cron: '0 0 * * *' },
+          timeout: { hours: 1 },
+          initialDelay: { hours: 1 },
+          scope: 'global',
+        };
+
+    const expiration: HumanDuration = config.has(
+      'notifications.expiration.duration',
+    )
+      ? readDurationFromConfig(config, {
+          key: 'notifications.expiration.duration',
+        })
+      : { years: 1 };
+    const expirationMs = durationToMilliseconds(expiration);
+    if (expirationMs > 0) {
+      const taskRunner = scheduler.createScheduledTaskRunner(schedule);
+      await taskRunner.run({
+        id: 'notification-cleaner',
+        fn: async () => {
+          await this.clearNotifications(logger, database, expirationMs);
```

Comment:
yeah ESPECIALLY using `this` here is kind of mind bending, if anything `NotificationCleaner. clearNotifications` would have been less surprising but java-ish